src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1148,6 +1148,61 @@ impl ToQemuParams for Rtc {
     }
 }
 
+#[derive(Debug)]
+struct RngDevice {
+    // id is the device ID
+    id: String,
+
+    // filename is the entropy source on the host
+    filename: String,
+
+    // max_bytes is the bytes allowed to guest to get from the host’s entropy per period
+    max_bytes: u32,
+
+    // period is duration of a read period in seconds
+    period: u32,
+
+    // transport is the virtio transport for this device.
+    transport: String,
+}
+
+impl RngDevice {
+    fn new() -> RngDevice {
+        RngDevice {
+            id: "rng0".to_owned(),
+            filename: "/dev/urandom".to_owned(),
+            max_bytes: 1024,
+            period: 1000,
+            transport: "virtio-rng-pci".to_owned(),
```

Comment:
Should this be `ccw` for s390x ? @BbolroC ?

Reply:
Yes, please. Thanks for the notice. 😉

Reply:
But, I've noticed that it is not instantiated for s390x. https://github.com/kata-containers/kata-containers/blob/64b915b86e5be1a2becd950644ec748123e50b1e/src/runtime/virtcontainers/qemu.go#L690-L701

Reply:
I think it should use `VirtioBusType` like other virtio devices.

Reply:
@BbolroC Ah okay, good catch, thanks!  @emanuellima1 So this should be reflected when the RNG is added (ie. only add it if not on a CCW system).